test/integration/use_test.rb

```diff
@@ -1,13 +1,59 @@
 require 'test_helper'
+require 'rack/test'
 
 describe 'Rack middleware integration' do
+  include Rack::Test::Methods
+
+  def response
+    last_response
+  end
+
   describe '.use' do
+    let(:app) { UseActionApplication }
+
     it 'uses the specified Rack middleware' do
-      response = Rack::MockRequest.new(UseAction.new).get('/')
-      status, headers, _ = response
+      router = Lotus::Router.new do
+        get '/', to: 'use_action#index'
+        get '/show', to: 'use_action#show'
+      end
+
+      UseActionApplication = Rack::Builder.new do
+        run router
+      end.to_app
+
+      get '/'
+
+      UseAction::Index.respond_to?(:call).must_equal true
```

Comment:
This is a low level detail that we can safely remove from here. The test already covers it. :cat: